packages/terra-form-validation/examples/index-examples/ValidationOnSubmit.jsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable class-methods-use-this */
+
+import React from 'react';
+import { Form, Field } from 'react-final-form';
+import InputField from 'terra-form-input/lib/InputField';
+import Button from 'terra-button';
+import Spacer from 'terra-spacer';
+
+const required = value => (value ? undefined : 'Required');
+
+export default class MainEntry extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  submitForm(values) {
+    this.setState({
+      submittedValues: values,
+    });
+  }
+
+  renderForm({ handleSubmit }) {
+    return (
+      <form
+        onSubmit={handleSubmit}
+      >
+        <Field
+          name="description"
+          validate={required}
+        >
+          {({ input, meta, placeholder, ...rest }) => (
+            <InputField
+              {...rest}
+              label="Description"
+              error={meta.error}
+              isInvalid={meta.submitFailed && meta.error}
+              inputAttrs={{
+                placeholder: 'Description',
+                required: true,
+                ...input,
+              }}
+              onChange={(e) => { input.onChange(e.target.value); }}
+              value={input.value}
+              required
+            />
+          )}
+        </Field>
+        <Button text="Submit" type={Button.Opts.Types.SUBMIT} />
```

Comment:
Let's include examples of validation for textarea, select, checkbox, and radio buttons as well in these demos.